autotest/gcore/data/vrt_subpixel_offset.vrt

```diff
@@ -0,0 +1,28 @@
+<VRTDataset rasterXSize="20" rasterYSize="20">
+  <SRS>PROJCS["NAD27 / UTM zone 11N",GEOGCS["NAD27",DATUM["North_American_Datum_1927",SPHEROID["Clarke 1866",6378206.4,294.9786982138982,AUTHORITY["EPSG","7008"]],AUTHORITY["EPSG","6267"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4267"]],PROJECTION["Transverse_Mercator"],PARAMETER["latitude_of_origin",0],PARAMETER["central_meridian",-117],PARAMETER["scale_factor",0.9996],PARAMETER["false_easting",500000],PARAMETER["false_northing",0],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AXIS["Easting",EAST],AXIS["Northing",NORTH],AUTHORITY["EPSG","26711"]]</SRS>
+  <GeoTransform>  4.4072000000000000e+05,  6.0000000000000000e+01,  0.0000000000000000e+00,  3.7513200000000000e+06,  0.0000000000000000e+00, -6.0000000000000000e+01</GeoTransform>
+  <VRTRasterBand dataType="Byte" band="1">
+    <ColorInterp>Gray</ColorInterp>
+    <SimpleSource resampling="cubic">
```

Comment:
@rouault I've been overlooking this resampling attribute... is it essential to use it? For example, if I open a VRT that has no simple source resampling, will calling `GDALRasterIO(..., resampling="cubic")` (pardon the shorthand) do the right thing?

Reply:
> f I open a VRT that has no simple source resampling, will calling `GDALRasterIO(..., resampling="cubic")` (pardon the shorthand) do the right thing?

yes. If the resampling attribute is set, it is used. Otherwise the user provided resampling method is used. See https://github.com/OSGeo/gdal/blob/master/gdal/frmts/vrt/vrtsources.cpp#L1287